### ISSUE TITLE
Implement Binance TradeLite message support

### ISSUE THREAD

**Issue by StephenLee2016** [nautechsystems/nautilus_trader#2008]
# Bug Report

### Expected Behavior
Add here...

### Actual Behavior
In live trading, there may be instances where trade_lite cannot be parsed. I noticed that previous submissions have added an “TRADE_LITE” enumeration field, but the parsing function was not found.

### Steps to Reproduce the Problem

[ERROR] TESTER-001.ExecClient-BINANCE: Error on handling b'{"stream":"jQ5bcqm7L6JDTIdGPx9NlWtxm2mkzRpVftCnxKrq817NEeDDUfdqHrC6vZck7A1w","data":{"e":"TRADE_LITE","E":1727683915920,"T":1727683915920,"s":"ETHUSDT","q":"0.037","p":"2631.90","m":false,"c":"O-20240930-081155-001-000-29","S":"BUY","L":"2631.90","l":"0.037","t":4467432841,"i":8389765739703625592}}'
KeyError(<BinanceFuturesEventType.TRADE_LITE: 'TRADE_LITE'>)
Traceback (most recent call last):
  File "/home/nautilus_trader/nautilus_trader/adapters/binance/futures/execution.py", line 277, in _handle_user_ws_message
    self._futures_user_ws_handlers[wrapper.data.e](raw)

### Specifications

- OS platform: Ubuntu
- Python version: 3.11
- `nautilus_trader` version: v1.203.0

**Comment by cjdsellers** [comment]
Hey @StephenLee2016

Thanks for the report!

I've implemented parsing for this message with this commit ac9dffa216d7cb823c9664de2883340c2df84c21.
So when this WS message is received it'll be parsed and delivered into [this handler](https://github.com/nautechsystems/nautilus_trader/blob/develop/nautilus_trader/adapters/binance/futures/execution.py#L303). Thats where the support currently ends though, as it needs to be further parsed into a `FillReport`.

I'm currently lacking the available bandwidth to continue to that.

Converting this ticket from bug to enhancement request to support the `TRADE_LITE` feature from Binance.
If someone else would like to pick this up then that would be much appreciated :pray:.

For myself, likely to be addressed later when we port this to Rust and implement the SBE protocol.

**Comment by DeevsDeevs** [comment]
Hey, @cjdsellers 

Can you assign this to me, please, i will try my best to finish this in a few days

**Comment by DeevsDeevs** [comment]
As i can see it right now:
Binance will push ORDER_TRADE_UPDATE of type TRADE event after TradeLite event, so nautilus will still close position.

While testing i didn't notice significant latency between this events, however TradeLite event won't have info about commissions.

So maybe we can add option to use TradeLite events instead of ORDER_TRADE_UPDATE to config of ```BinanceCommonExecutionClient```.

What do you think?

**Comment by cjdsellers** [comment]
Hey @DeevsDeevs

Looking more closely at this, the `TRADE_LITE` messages probably aren't suitable for Nautilus, because we do need to record the commissions for the trade.

It's probably better to explicitly state in the docs that the `TRADE_LITE` feature isn't supported?

**Comment by sunlei** [comment]
@cjdsellers The same issue exists with Bybit's `execution.fast`.

**Comment by DeevsDeevs** [comment]
@cjdsellers i think that commission part can be optional

there is already code:
```
  commission_asset: str | None = self.N
  commission_amount: str | None = self.n
  if commission_asset is not None:
      commission = Money.from_str(f"{commission_amount} {commission_asset}")
  else: # (same as TradeLite)
      # Commission in margin collateral currency
      commission = Money(0, instrument.quote_currency)
```

so if user intentionally turned on TradeLite in config, we will work with this type of events for fills without commissions

I don't really know use-case for this stream, but we can give this opportunity imho

**Comment by cjdsellers** [comment]
We could also calculate commission from known fee rates. Maybe @sunlei has an opinion here for Bybit, in practice?

> Fast execution stream significantly reduces data latency compared original "execution" stream

Do we have an indication of what the perf improvements of these message types actually are?

**Comment by sunlei** [comment]
@cjdsellers 

Bybit should also be able to calculate based on fees.
But there’s another issue with Bybit: in `execution.fast`, the `orderLinkId` for maker trades is always an empty string ("").
So, would the following approach work?
When generating the report, retrieve the order using the venue order ID, and then obtain the client order ID from the order.

From our experience with Binance TradeLite, this could save at most a few hundred microseconds.
As for Bybit, this hasn’t been tested yet in practice.

**Comment by cjdsellers** [comment]
> So, would the following approach work? When generating the report, retrieve the order using the venue order ID, and then obtain the client order ID from the order.

Yes, I think substituting any missing info by using the cache would be fine in this case.

**Comment by DeevsDeevs** [comment]
So I will try to implement calculating fees for TradeLite events inside this handler based on cache fees for account
But still I think config field is needed to select whether to use TradeLite or just default ORDER_UPDATE events

**Comment by cjdsellers** [comment]
Closed from #2156 many thanks @DeevsDeevs
